File: be/src/models/person.rs
use chrono::NaiveDateTime;
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Person {
    pub id: Uuid,
    pub first_name: String,
    pub middle_name: String,
    pub last_name: String,
    pub created_at: NaiveDateTime,
}

File: fe/src/lib/types/person.ts
export interface Person {
    id: string;
    first_name: string;
    middle_name?: string;
    last_name: string;
    email?: string;
    phone?: string;
    date_of_birth?: string;
    citizenship_number?: string;
    created_at: string;
}

export interface CreatePersonRequest {
    first_name: string;
    middle_name?: string;
    last_name: string;
    email?: string;
    phone?: string;
}

export interface UpdatePersonRequest {
    first_name?: string;
    middle_name?: string;
    last_name?: string;
    email?: string;
    phone?: string;
}

